packages/base/src/utils/time-range.ts

```diff
@@ -1,18 +1,47 @@
+export interface TimeRangeString {
+    start: string;
+    end: string;
+    offset?: string;
+}
+
 export class TimeRange {
     private start: bigint;
     private end: bigint;
     private offset: bigint | undefined;
 
+    /**
+     * Constructor.
+     * Default TimeRange with 0 for values
+     */
+    constructor();
     /**
      * Constructor.
      * @param start Range start time
      * @param end Range end time
      * @param offset Time offset, if this is defined the start and end time should be relative to this value
      */
-    constructor(start: bigint, end: bigint, offset?: bigint) {
-        this.start = start;
-        this.end = end;
-        this.offset = offset;
+    constructor(timeRangeString: TimeRangeString);
+    /**
+     * Constructor.
+     * @param timeRangeString string object returned by this.toString()
+     */
```

Comment:
Need to swap the last two Javadocs

Reply:
Or swap the constructors. Probably makes more sense to have the one with (start, end, offset) before the one with (TimeRangeString).

Reply:
Oops, sorry!

Reply:
@PatrickTasse I put the JavaDocs in the correct spot, and moved the (start, end offset) constructor to the first position.